portfolio/src/main/java/com/google/sps/servlets/DeleteAllDataServlet.java

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for deleting comments. */
+@WebServlet("/delete-all-comments")
+public class DeleteAllDataServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("message", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId(); 
+        Key commentEntityKey = KeyFactory.createKey("Comment", id);
+        datastore.delete(commentEntityKey);
+        System.out.println("Deleted object with id: " + id);
+    }
```

Comment:
There is a batch option to delete multiple keys at once [1], [2]. This would be more efficient since it only makes one call to the datastore service.

You can use Iterables.transform [3] to transform each entity to its key. Note, you'll have to add guava to your pom file.

Iterable<Key> keysToDelete = Iterables.transform(results.asIterable(), Entity::getKey); datastore.delete(keysToDelete);

[1] https://cloud.google.com/appengine/docs/standard/java/datastore/creating-entities#Batch_operations
[2] https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html#delete_2
[3] https://guava.dev/releases/21.0/api/docs/com/google/common/collect/Iterables.html#transform-java.lang.Iterable-com.google.common.base.Function-